pkg/cmd/pulumi/destroy.go

```diff
@@ -43,6 +44,19 @@ func newDestroyCmd() *cobra.Command {
 	var execKind string
 	var execAgent string
 
+	// Flags for remote operations.
+	var remote bool
+	var envVars []string
+	var preRunCommands []string
+	var gitBranch string
+	var gitCommit string
+	var gitRepoDir string
+	var gitAuthAccessToken string
+	var gitAuthSSHPrivateKey string
+	var gitAuthSSHPrivateKeyPath string
+	var gitAuthPassword string
+	var gitAuthUsername string
```

Comment:
It looks like this appears in destroy, up, preview and refresh. I suspect we will always want these variables to be the same across commands. 
To avoid leaking how remote works (and risking inconsistencies across commands), we can encapsulate these in a struct:
```go
// Flags for remote operations.
type RemoteArgs struct {
  remote bool
  envVars []string
  preRunCommands []string
  gitBranch string
  ...
}

// Add flags to support remote operations
func (r *RemoteArgs) applyFlags(cmd *cobra.Command) {
    cmd.PersistentFlags().BoolVar(
		&r.remote, "remote", false,
		"[EXPERIMENTAL] Run the operation remotely")
  cmd.PersistentFlags().StringArrayVar(
		&r.envVars, "remote-env", []string{},
		"[EXPERIMENTAL] Environment variables to use in the remote operation of the form NAME=value for "+
			"plaintext values or NAME#=value for secret values (e.g. `--remote-env FOO=bar "+
			"--remote-env BAR#=secret`)")
  ...
}
```

and then initialize them like so:

```go
// With the other variables
remoteArgs := RemoteArgs{}
...

// With the other flags
remoteArgs.applyFlags(cmd)
```